httpbp/middlewares.go

```diff
@@ -108,6 +110,14 @@ func StartSpanFromTrustedRequest(
 	return tracing.StartSpanFromHeaders(ctx, name, spanHeaders)
 }
 
+func HTTPErrorSuppressor(err error) bool {
```

Comment:
Please add doc comment for this function.

Reply:
This doc comment just says what it does, not why I would use it.

"HTTPErrorSuppressor causes metrics to ignore non-5xx errors.  This should be used when dashboards
relying on the builtin metrics assume that client-driven failure modes are excluded."

or something.

That being said, what about just having a new middleware that exposes that 5xx-rate separately, so that we expose both, and then dashboards can use whichever makes the most sense for the service?

Reply:
> That being said, what about just having a new middleware that exposes that 5xx-rate separately, so that we expose both, and then dashboards can use whichever makes the most sense for the service?

We could (and it's probably a good idea to) add that, but I think we still want this since this affects what get's reported as a "success" or "failure" in the baseplate endpoint metrics.

Reply:
maybe we should also just un-export this for now since it really isn't something you would use currently, we would export it if we had you pass in an `errorsbp.Suppressor` though.

Reply:
yea +1 to unexport it. for http errors server errors and client errors are also different (and this suppressor only works for server errors) which makes the public api more complicated. we can export it later when needed.